File: tempo-api/src/app/tempo/models/song_post.py
from . import *

class SongPost(Base):
  __tablename__ = 'song_posts'

  id = db.Column(db.Integer, primary_key=True)
  post_id = db.Column(db.Integer, db.ForeignKey('posts.id', ondelete='CASCADE'))
  song_id = db.Column(db.Integer, db.ForeignKey('songs.id', ondelete='CASCADE'))

  song = db.relationship('Song', cascade='all,delete')

  def __init__(self, **kwargs):
    self.post_id = kwargs.get('post_id', 0)
    self.song_id = kwargs.get('song_id', 0)

File: tempo-backend/app/models/song_post.rb
# == Schema Information
#
# Table name: song_posts
#
#  id         :integer          not null, primary key
#  post_id    :integer
#  song_id    :integer
#  created_at :datetime         not null
#  updated_at :datetime         not null
#

class SongPost < ActiveRecord::Base
  belongs_to :posts
  belongs_to :songs
end
